### PR TITLE
Add example with a communication thread.

### PR DESCRIPTION
**Pull request by windelbouwman**
As discussed [here](https://github.com/gtk-rs/glib/issues/584), this works very sweet!

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Thanks for the new example! Before reviewing/merging, please target the pending branch (switching here doesn't give a nice result so I'll let you handle it on your side).

File: src/bin/communication_thread.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    // Spawn queue receive task on the main event loop:
+    let main_context = glib::MainContext::default();
+    let future = async move {
+        while let Some(item) = receiver.next().await {
+            label.set_text(&item);
+        }
+    };
+    main_context.spawn_local(future);
+
+    // Spawn seperate thread to handle communication:
```
Comment (EPashkin) [inline src/bin/communication_thread.rs]: Typo: sep`a`rate

**Comment by EPashkin** [comment]
@windelbouwman LGFM, only one typo and wrong base branch

**Comment by windelbouwman** [comment]
Please refer to #290

**Pull request closed without merging**